packages/core-accountsjs/accounts-server.js

```diff
@@ -9,6 +9,15 @@ const accountsServerOptions = {
   siteUrl: process.env.ROOT_URL,
   prepareMail: (to, token, user, pathFragment, emailTemplate, from) => {
     if (token && pathFragment) {
+      // we're not supposed to be sending verifications to guests
+      // nor send verification emails to just **passwordless** enrolled users
+      if (
+        (user.guest || !user?.services?.password?.bcrypt) &&
```

Comment:
If (pathFragment === "verify-email" && user.isInitialPassword()) return {
        recipientEmail: to,
        action: 'enrollAccount',
        userId: user.id || user._id,
        token,
      };

Reply:
This relates to the previous comment. We need to get in the way of verify-email which is fired automatically yet let enroll-accounts be sent freely without any interference.

Reply:
I'm not sure if you can use the current implementation of isInitialPassword here so please check carefully.